packages/okta-adapter/src/config.ts

```diff
@@ -0,0 +1,712 @@
+    { typeName: 'OAuthAuthorizationPolicies', cloneFrom: 'api__v1__policies' },
+    // TODO this is not the right type to clone from, another solution is needed
+    { typeName: 'RolePage', cloneFrom: 'api__v1__groups___groupId___roles@uuuuuu_00123_00125uu' },
+  ],
+}
+
+/* eslint-disable max-len */
+export const SUPPORTED_TYPES = {
+  Application: [
+    'api__v1__apps',
+  ],
+  IdentityProviderJsonWebKey: [
+    'api__v1__idps__credentials__keys',
+  ],
+  Authenticator: ['api__v1__authenticators'],
+  AuthorizationServer: ['api__v1__authorizationServers'],
+  Brand: ['api__v1__brands'],
+  EventHook: ['api__v1__eventHooks'],
+  Feature: ['api__v1__features'],
+  Group: [
+    'api__v1__groups',
+  ],
+  User: [
+    'api__v1__users',
+  ],
+  GroupRule: ['api__v1__groups__rules'],
+  IdentityProvider: [
+    'api__v1__idps',
+  ],
+  InlineHook: ['api__v1__inlineHooks'],
+  // TODO returns 401
+  ProfileMapping: ['api__v1__mappings'],
+  LinkedObjectDefinitions: ['api__v1__meta__schemas__user__linkedObjects'],
+  GroupSchema: ['GroupSchema'],
+  UserSchema: ['UserSchema'],
+  UserType: ['api__v1__meta__types__user'],
+  OrgContactTypeObj: ['api__v1__org__contacts'],
+  OrgSettings: ['OrgSetting'],
+  Policy: [
+    'AuthenticatorEnrollmentPolicies',
+    'GlobalSessionPolicies',
+    'AuthenticationPolicies',
+    'ProfileEnrollmentPolicies',
+    'IdentityProviderRoutingRules',
+    'PasswordPolicies',
+    'OAuthAuthorizationPolicies',
+  ],
+  SmsTemplate: ['api__v1__templates__sms'],
+  TrustedOrigin: ['api__v1__trustedOrigins'],
+  NetworkZone: ['api__v1__zones'],
+  Domain: ['DomainListResponse'],
+  Role: ['RolePage'],
+}
+
+
+export const DEFAULT_API_DEFINITIONS: OktaApiConfig = {
+  swagger: DEFAULT_SWAGGER_CONFIG,
+  typeDefaults: {
+    transformation: {
+      idFields: DEFAULT_ID_FIELDS,
+      fieldsToOmit: FIELDS_TO_OMIT,
+      // TODO: currently these fields can be nested
+      // fieldsToHide: [
+      //   { fieldName: 'created', fieldType: 'string' },
+      //   { fieldName: 'lastUpdated', fieldType: 'string' },
+      //   { fieldName: 'lastLogin', fieldType: 'string' },
+      //   { fieldName: '_links', fieldType: 'string' },
+      // ],
+    },
+  },
+  types: DEFAULT_TYPE_CUSTOMIZATIONS,
+  supportedTypes: SUPPORTED_TYPES,
+}
+
+export const DEFAULT_CONFIG: OktaConfig = {
+  [FETCH_CONFIG]: elements.query.INCLUDE_ALL_CONFIG,
+  [API_DEFINITIONS_CONFIG]: DEFAULT_API_DEFINITIONS,
+}
+
+export const configType = createMatchingObjectType<Partial<OktaConfig>>({
+  elemID: new ElemID(OKTA),
+  fields: {
+    [CLIENT_CONFIG]: {
+      refType: createClientConfigType(OKTA),
+    },
+    [FETCH_CONFIG]: {
+      refType: createUserFetchConfigType(
+        OKTA,
+      ),
+    },
+    [API_DEFINITIONS_CONFIG]: {
+      refType: createSwaggerAdapterApiConfigType({
+        adapter: OKTA,
+      }),
+    },
+  },
+  annotations: {
+    [CORE_ANNOTATIONS.DEFAULT]: _.omit(DEFAULT_CONFIG, API_DEFINITIONS_CONFIG),
+  },
```

Comment:
we should also add this now so we can enforce there are only supported configuration properties (this is a new feature that was added after you started the adapter) - here and anywhere else we create a config object type (I think this is the only one for now):
```suggestion
    [CORE_ANNOTATIONS.ADDITIONAL_PROPERTIES]: false,
  },
```